helpers/getTestimonials.ts

```diff
@@ -1,40 +1,103 @@
+  name: string;
+  avatar?: string;
+  company?: string;
+  body?: string;
+};
+const SSWInternalTestimonialCategories = ["internship", "brainstorming"];
+
+export const getTestimonialsByCategories = async (
+  categories: string[] = [],
+  isOrdered: boolean = true
+): Promise<TestimonialType[] | []> => {
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some((testimonialCategory) =>
+          categories.length > 0
+            ? categories.some(
+                (category) =>
+                  category === extractFileName(testimonialCategory.category)
+              )
+            : extractFileName(testimonialCategory.category) === "General"
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType);
+
+  const testimonials = isOrdered
+    ? testimonialsResult
+    : testimonialsResult.sort(() => 0.5 - Math.random());
+
+  return testimonials?.slice(0, 3);
+};
+
+export const getFilteredTestimonials = async (
+  categoryRoutePaths: {
+    categoryName: string;
+  }[]
+): Promise<TestimonialType[]> => {
+  const categoryListToExclude = categoryRoutePaths?.map(
+    (category) => extractFileName(category?.categoryName)?.toLowerCase()
+  );
+
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some(
+          (testimonialCategory) =>
+            categoryListToExclude.length > 0 &&
+            !categoryListToExclude.some(
+              (category) =>
+                category ===
+                extractFileName(testimonialCategory.category)?.toLowerCase()
+            )
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType);
+
+  return testimonialsResult;
+};
+export const getRandomClientTestimonial = async () => {
+  const testimonialsResult = testimonialList.testimonials
+    .filter(
+      (testimonial) =>
+        testimonial.categories &&
+        testimonial.categories.some(
+          (testimonialCategory) =>
+            SSWInternalTestimonialCategories.length > 0 &&
+            !SSWInternalTestimonialCategories.some(
+              (category) =>
+                category ===
+                extractFileName(testimonialCategory.category)?.toLowerCase()
+            )
+        )
+    )
+    .map((testimonial) => testimonial as TestimonialType);
+
+  const randomTestimonial =
+    testimonialsResult[Math.floor(Math.random() * testimonialsResult.length)];
+
+  return toStringify(randomTestimonial);
+};
+
+export const testimonialToSelectOptions = () => {
+  const testimonialOptions = testimonialList.testimonials.map(
+    (testimonial) => ({
+      value: toStringify(testimonial),
+      label: `${testimonial.name} - ${testimonial.company ?? ""}`,
+    })
+  );
+
+  return testimonialOptions;
+};
+
+const toStringify = (testimonial: TestimonialType) =>
```

Comment:
the usage of this method seems very odd - why are we passing around json strings instead of the typescript objects?

Reply:
done - removed this function and now passing this as an object